pom.xml

```diff
@@ -42,12 +42,42 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
   </properties>
 
+  <profiles>
+    <profile>
+      <id>release-sign-artifacts</id>
+      <activation>
+        <property>
+          <name>performRelease</name>
+          <value>true</value>
+        </property>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-gpg-plugin</artifactId>
+            <version>3.0.1</version>
+            <executions>
+              <execution>
+                <id>sign-artifacts</id>
+                <phase>verify</phase>
+                <goals>
+                  <goal>sign</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+  </profiles>
+
   <build>
     <plugins>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-source-plugin</artifactId>
-        <version>2.2.1</version>
+        <version>3.2.1</version>
```

Comment:
Change log is too long to read (128 commits) but it seems it contains mostly infrastructure changes. I think the best way is just check the output -- it shouldn't be too hard. At least it generates doc and they seems correct.

The most important thing in 2.x -> 3.x migration seems to be summarized in this phrase from the [plugin's page](https://maven.apache.org/plugins/maven-source-plugin/):
> Starting with version 3.0.0 of the plugin all properties which could be used via command line have been named based on the following schema `maven.source.*`. Further details can be found in the goal documentations.

As I see we don't use CLI, so that does not influence us.

Reply:
@generalmimon, could you comment on this?